hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieClient.java

```diff
@@ -284,4 +288,33 @@ protected void writeTableMetadata(HoodieTable table, String instantTime, HoodieC
       }
     }
   }
+
+  public static void updateColsToIndex(HoodieTable dataTable, HoodieWriteConfig config, HoodieCommitMetadata commitMetadata) {
+    if (config.getMetadataConfig().isColumnStatsIndexEnabled()) {
+      try {
+        HoodieTableMetaClient mdtMetaClient = HoodieTableMetaClient.builder()
+            .setStorage(dataTable.getStorage())
+            .setBasePath(HoodieTableMetadata.getMetadataTableBasePath(dataTable.getMetaClient().getBasePath()))
+            .build();
+        HoodieInstant latestInstant = mdtMetaClient.getActiveTimeline().filterCompletedInstants().getInstantsOrderedByCompletionTime().reduce((a,b) -> b).get();
+
+        final HoodieCommitMetadata mdtCommitMetadata = mdtMetaClient.getTimelineLayout().getCommitMetadataSerDe().deserialize(
+            latestInstant,
+            mdtMetaClient.getActiveTimeline().getInstantDetails(latestInstant).get(),
+            HoodieCommitMetadata.class);
+        if (mdtCommitMetadata.getPartitionToWriteStats().containsKey(MetadataPartitionType.COLUMN_STATS.getPartitionPath())) {
```

Comment:
Another issue is that the check is costly. If not for this check, then we don't need `mdtCommitMetadata` and hence don't need to create `mdtMetaClient`. If we strictly need this check, please consider caching the meta client or reusing an existing instance if possible.